aten/src/ATen/native/MaxUnpooling.cpp

```diff
@@ -0,0 +1,615 @@
+#include <tuple>
+#include "ATen/ATen.h"
+#include "ATen/NativeFunctions.h"
+
+namespace at {
+namespace native {
+
+template <typename scalar_t>
+Tensor MaxUnpooling2d_forward_out_cpu_(
+    Tensor& output,
+    const Tensor& input,
+    const Tensor& indices,
+    int64_t outputHeight,
+    int64_t outputWidth) {
+  int64_t numBatch = 1;
+  int64_t dimc = 0;
+  int64_t dimh = 1;
+  int64_t dimw = 2;
+  if (input.ndimension() == 4) {
+    numBatch = input.size(0);
+    dimc++;
+    dimh++;
+    dimw++;
+  }
+  auto numChannels = input.size(dimc);
+  auto inputHeight = input.size(dimh);
+  auto inputWidth = input.size(dimw);
+
+  auto* rawInput = input.data<scalar_t>();
+  auto* rawIndices = indices.data<int64_t>();
+  auto* rawOutput = output.data<scalar_t>();
+
+  for (auto n = 0; n < numBatch; n++) {
```

Comment:
"auto n = 0;" is almost certainly not what you want; you want n to be the same type as numBatch.